flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/dialect/JDBCDialects.java

```diff
@@ -211,6 +240,28 @@ public int minTimestampPrecision() {
 					LogicalTypeRoot.SYMBOL,
 					LogicalTypeRoot.UNRESOLVED);
 		}
+
+		@Override
+		public String getDialectTypeName(DataType dataType) {
+			switch (dataType.getLogicalType().getTypeRoot()) {
+				case VARCHAR:
+					final int len = ((VarCharType) dataType.getLogicalType()).getLength();
+					return String.format("VARCHAR(%d)", len > MAX_VARCHAR_LEN ? MAX_VARCHAR_LEN : len);
+				case DECIMAL:
+					if (dataType.getLogicalType() instanceof DecimalType) {
+						return dataType.toString();
+					}
+					// for legacy type
+					return "DECIMAL(" + MAX_DECIMAL_PRECISION + ", 18)";
+				// derby does not support time/timestamp type with precision
+				case TIME_WITHOUT_TIME_ZONE:
+					return "TIME";
+				case TIMESTAMP_WITHOUT_TIME_ZONE:
+					return "TIMESTAMP";
+				default:
+					return dataType.toString();
```

Comment:
Let's list the full mapping, not fallback to the default toString, this is error-prone. 
You can have a look at `PostgresCatalog#fromJDBCType`.